src/main/java/org/jenkinsci/plugins/vSphereCloudSlaveTemplate.java

```diff
@@ -175,8 +175,8 @@ public vSphereCloudSlaveTemplate(final String cloneNamePrefix,
         this.targetResourcePool = targetResourcePool;
         this.targetHost = targetHost;
         this.credentialsId = credentialsId;
-        this.nodeProperties = nodeProperties;
-        this.guestInfoProperties = guestInfoProperties;
+        this.nodeProperties = Util.fixNull(nodeProperties);
+        this.guestInfoProperties = Util.fixNull(guestInfoProperties);
```

Comment:
While I'm kind-of ok with ensuring `nodeProperties` is an empty list instead of a null one, this isn't necessary for `guestInfoProperties` as the code which uses this copes with nulls.

...and if you want to ensure that either of these are non-null then we'll also need code in `readResolve()` to fix nulls there, as the constructor is only called when new instances are created; it isn't called when the instance gets de-serialized from the XML in config.xml
Personally, I quite like classes that get serialized as XML to have fields set to `null` when there's nothing in there (and then having the code handle nulls) as that way the serialized form of the class is smaller _and_ it makes the upgrade path easier, but I appreciate that it can make the code harder.
Either way, it's not a job for this PR; either another "generic code tidy-up" PR or not at all.

Reply:
This is needed to be JCasc compliant, and I would rather NOT deal with potential null pointers. I feel it is safer be checking for empty lists.

Reply:
When going via stapler and UI the node properties are definitely set to a empty list for you. But when JCasc configures the plugin it leaves them as null and then you should get null pointers somewhere down in the Slave/Computer object.

Reply:
Ah, OK, if this is required for JCasc then it can stay in this PR.
I don't think it's necessary to do this for `guestInfoProperties` as that code is fully null-proof, but if it's needed for `nodeProperties` then I guess that line is needed.
I would _prefer_ guestInfoProperties to be null if possible, as that means that when it's turned into XML there is no `<guestInfoProperties />` element, but if JCasc requires an empty list there as well then I guess it'll have to be `fixNull`ed too.

Reply:
Still needed otherwise user will experience null pointer exception when applying JCasC configuration